src/figuresClasses.ts

```diff
@@ -1,19 +1,84 @@
 export interface Figure {
-
+  shape: string;
+  color: string;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: string = 'triangle';
 
-}
+  color: string;
 
-export class Circle {
+  constructor(
+    color: string,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0 || a + b <= c || a + c <= b || b + c <= a) {
+      throw new Error('your error message');
+    }
+    this.color = color;
+  }
 
+  getArea(): number {
+    const { a, b, c } = this;
+    const halfPerimeter = (a + b + c) / 2;
+    const area = Math.sqrt(halfPerimeter
+      * (halfPerimeter - a) * (halfPerimeter - b) * (halfPerimeter - c))
+      .toFixed(2);
+
+    return +area;
+  }
 }
 
-export class Rectangle {
+export class Circle implements Figure {
+  shape: string = 'circle';
+
+  color: string;
+
+  constructor(
+    color: string,
+    public radius: number,
+  ) {
+    if (radius <= 0) {
+      throw new Error('your error message');
+    }
+
+    this.color = color;
+  }
+
+  getArea(): number {
+    const { radius } = this;
 
+    return Math.floor(Math.PI * (radius ** 2) * 100) / 100;
+  }
 }
 
-export function getInfo(figure) {
+export class Rectangle implements Figure {
+  shape: string = 'rectangle';
```

Comment:
please move 'rectangle', 'circle' and 'triangle' to the enum as it will make a code cleaner